tests/test_internals/test_movie_utils.py

```diff
@@ -0,0 +1,78 @@
+import os
+import tempfile
+import unittest
+import cv2
+import numpy as np
+from numpy.testing import assert_array_equal
+from nwb_conversion_tools.datainterfaces.behavior.movie.movie_utils import VideoCaptureContext
+
+
+class TestVideoContext(unittest.TestCase):
+
+    frame_shape = (100, 200, 3)
+    no_frames = 30
+    fps = 25
+
+    def setUp(self) -> None:
+        self.test_dir = tempfile.mkdtemp()
+        self.movie_frames = np.random.randint(0, 255, size=[self.no_frames, *self.frame_shape], dtype="uint8")
+        self.movie_loc = self.create_movie()
+
+    def create_movie(self):
+        movie_file = os.path.join(self.test_dir, "test.avi")
+        writer = cv2.VideoWriter(
+            filename=movie_file,
+            apiPreference=None,
+            fourcc=cv2.VideoWriter_fourcc(*"HFYU"),
+            fps=self.fps,
+            frameSize=self.frame_shape[1::-1],
+            params=None,
+        )
+        for k in range(self.no_frames):
+            writer.write(self.movie_frames[k, :, :, :])
+        writer.release()
+        return movie_file
+
+    def test_context(self):
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            pass
+        self.assertFalse(vcc.vc.isOpened())
+
+    def test_stub(self):
+        with VideoCaptureContext(self.movie_loc, stub=True) as vcc:
+            frame_count = vcc.get_movie_frame_count()
+        self.assertEqual(frame_count, 10)
+
+    def test_timestamps(self):
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            ts = vcc.get_movie_timestamps()
+        self.assertEqual(len(ts), self.no_frames)
+        with VideoCaptureContext(self.movie_loc, stub=True) as vcc:
+            ts = vcc.get_movie_timestamps()
+        self.assertEqual(len(ts), 10)
+
+    def test_fps(self):
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            fps = vcc.get_movie_fps()
+        self.assertEqual(fps, self.fps)
+
+    def test_frame_shape(self):
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            frame_shape = vcc.get_frame_shape()
+        assert_array_equal(frame_shape, self.frame_shape)
+
+    def test_frame_value(self):
+        frames = []
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            for no in range(self.no_frames):
+                frames.append(vcc.get_movie_frame(no))
+        assert_array_equal(frames, self.movie_frames)
+
+    def test_iterable(self):
+        frames = []
+        with VideoCaptureContext(self.movie_loc) as vcc:
+            pass
+        for frame in vcc:
+            frames.append(frame)
+        assert_array_equal(np.array(frames), self.movie_frames)
```

Comment:
```suggestion
        with VideoCaptureContext(self.movie_loc) as vcc:
            pass
            for frame in vcc:
                frames.append(frame)
        assert_array_equal(np.array(frames), self.movie_frames)
```

Did you intend this kind of nesting here? It's otherwise rather odd to alias the context to a variable which then gets referenced outside the context closure.

Reply:
Or maybe this is meant to just be

```python
        for frame in VideoCaptureContext(file_path=self.movie_loc):
            frames.append(frame)
        assert_array_equal(np.array(frames), self.movie_frames)
```

Reply:
I want the Context class to be an iterable even if the `VideoCapture` object is closed. This is what this line is doing: it opening and then closing the `VideoCapture` object. Irrespective, we can still iterate over it. 
But I have reduced it here https://github.com/catalystneuro/nwb-conversion-tools/pull/392/commits/42141abd2d89a53904f663d0f9bc94e3a3002893